### PR TITLE
Linux build instructions in `BUILD.md`

### LINKED ISSUE #501: Add Linux compile instructions
Thanks to @itsstepigo, @elekius has blessed us with a Dockerfile for building Northstar. We should integrate this into some docs (`BUILD.md` or other).

Download the [Dockerfile.txt](https://github.com/R2Northstar/NorthstarLauncher/files/11956764/Dockerfile.txt) and rename it just `Dockerfile` (GitHub wouldn't allow uploading otherwise).

The run the following commands:

1. Build image:
   ```sh
   docker build --rm -t test-fedora .
   ```

2. Then go to NorthstarLauncher folder and run:
- ```sh
   docker run --rm -it -e CC=cl -e CXX=cl --mount type=bind,source="$(pwd)",destination=/build test-fedora cmake . -DCMAKE_BUILD_TYPE=Release -DCMAKE_SYSTEM_NAME=Windows -G "Ninja"
   ```
- ```sh
   docker run --rm -it -e CC=cl -e CXX=cl --mount type=bind,source="$(pwd)",destination=/build test-fedora cmake --build .
   ```

### PR DESCRIPTION
**Pull request by itscynxx**
Adds the instructions given in #501 to `BUILD.md` for building the launcher on Linux. 

Closes #501

### CODE REVIEW

File: BUILD.md [hunk 1]
```diff
@@ -2,7 +2,7 @@
 The following steps will allow you to compile your own NorthstarLauncher executable from the code in this repository. If you still have questions, you may ask in the [Discord Server](https://discord.gg/northstar)
 
 *This guide assumes you have already installed Northstar as shown in [this page](https://r2northstar.gitbook.io/r2northstar-wiki/installing-northstar/basic-setup)*
-
+## Windows
```
Comment (GeckoEidechse) [inline BUILD.md:5]: Missing newline
```suggestion

## Windows
```

File: BUILD.md [hunk 2]
```diff
@@ -39,3 +40,13 @@ Developers who can work a command line may be interested in using [Visual Studio
 - Run `cmake . -G "Ninja"` to generate build files.
 
 - Run `cmake --build .` to build the project.
+  
+## Linux
+### Steps
+1. Download the [Dockerfile.txt](https://github.com/R2Northstar/NorthstarLauncher/files/11956764/Dockerfile.txt) file. Rename it to just `Dockerfile` (GitHub wouldn't allow uploading otherwise).
```
Comment (Alystrasz) [inline BUILD.md:46]: Why not directly adding the Dockerfile to the repository?
The one [mentioned here](https://github.com/R2Northstar/NorthstarLauncher/issues/501#issuecomment-1666499716) seems to be working at least on Ubuntu.

**Review by Alystrasz — CHANGES_REQUESTED** [review]

File: BUILD.md (hunk 2 above)
Comment (CooldudePUGS) [reply BUILD.md:46]: Adding it is fine, any idea where I should add it to keep things cleaner? I'd imagine root directory is probably the best place for it (so alongside `BUILD.md`)

Comment (Alystrasz) [reply BUILD.md:46]: Root dir is cool yeah

Comment (CooldudePUGS) [reply BUILD.md]: done

Comment (Alystrasz) [reply BUILD.md]: You should remove the .txt file extension + your first bullet point

Comment (CooldudePUGS) [reply BUILD.md]: I guess this does work, I was just told github wouldnt let upload the file without the .txt when i was writing it

**Review by Alystrasz — CHANGES_REQUESTED** [review]

File: BUILD.md [hunk 3]
```diff
@@ -39,3 +40,12 @@ Developers who can work a command line may be interested in using [Visual Studio
 - Run `cmake . -G "Ninja"` to generate build files.
 
 - Run `cmake --build .` to build the project.
+  
+## Linux
+### Steps
+1. Download the [Dockerfile](Dockerfile) file.
```
[the remark this answers is not in the document]
Comment (CooldudePUGS) [reply BUILD.md]: forgot to do this, oops

also added some clarification to more closely match the windows instructions

File: BUILD.md [hunk 4]
```diff
@@ -39,3 +40,13 @@ Developers who can work a command line may be interested in using [Visual Studio
...
 
 - Run `cmake --build .` to build the project.
+  
+## Linux
+### Steps
+1. Clone the GitHub repo
+2. Use `cd` to navigate to the cloned repeo's directory
+3. Run the following command:
+* `docker build --rm -t test-fedora`
+4. Go to the `NorthstarLauncher` folder inside the cloned repo and run the following commands:
```
Comment (Alystrasz) [inline BUILD.md:50]:
```suggestion
```
All commands need to be run from root directory

**Review by Alystrasz — CHANGES_REQUESTED** [review]

**Comment by ASpoonPlaysGames** [comment]
(Testing would just be following the instructions and making sure they work ig)

File: Dockerfile [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+	python3-simplejson \
+	python3-six \
+	cmake \
+	ninja-build \
+	make \
+	samba \
+	libunwind && \
+    dnf clean all && \
+    mkdir /opt/msvc/ /build
+
```
Comment (uniboi) [inline Dockerfile:18]: doesn't work without this for me
```suggestion

RUN git config --global --add safe.directory /build
```

File: BUILD.md [hunk 6]
```diff
@@ -39,3 +40,12 @@ Developers who can work a command line may be interested in using [Visual Studio
 - Run `cmake . -G "Ninja"` to generate build files.
 
 - Run `cmake --build .` to build the project.
+  
+## Linux
+### Steps
+1. Clone the GitHub repo
+2. Use `cd` to navigate to the cloned repeo's directory
```
Comment (Alystrasz) [inline BUILD.md:47]:
```suggestion
2. Use `cd` to navigate to the cloned repo's directory
```

File: Dockerfile (hunk 5 above)
Comment (Alystrasz) [reply Dockerfile:18]: I had to do the same iirc

**Review by uniboi — APPROVED** [review]
The process worked for me

File: BUILD.md [hunk 7]
```diff
@@ -39,3 +40,12 @@ Developers who can work a command line may be interested in using [Visual Studio
...
 - Run `cmake --build .` to build the project.
+  
+## Linux
+### Steps
+1. Clone the GitHub repo
+2. Use `cd` to navigate to the cloned repo's directory
+3. Then, run the following commands in order:
+* `docker build --rm -t test-fedora .`
+* `docker run --rm -it -e CC=cl -e CXX=cl --mount type=bind,source="$(pwd)",destination=/build test-fedora cmake . -DCMAKE_BUILD_TYPE=Release -DCMAKE_SYSTEM_NAME=Windows -G "Ninja"`
+* `docker run --rm -it -e CC=cl -e CXX=cl --mount type=bind,source="$(pwd)",destination=/build test-fedora cmake --build .`
```
Comment (GeckoEidechse) [inline BUILD.md:51]: I would rename the image from `test-fedora` to something like `northstar-build-fedora` or something along the lines ^^

Comment (uniboi) [reply BUILD.md:51]:
```suggestion
* `docker build --rm -t northstar-build-fedora .`
* `docker run --rm -it -e CC=cl -e CXX=cl --mount type=bind,source="$(pwd)",destination=/build northstar-build-fedora cmake . -DCMAKE_BUILD_TYPE=Release -DCMAKE_SYSTEM_NAME=Windows -G "Ninja"`
* `docker run --rm -it -e CC=cl -e CXX=cl --mount type=bind,source="$(pwd)",destination=/build northstar-build-fedora cmake --build .`
```
make a suggestion then

**Review by GeckoEidechse — APPROVED** [review]
Updated image name confirmed working in testing.

Compiled fine and resulting binary ran just fine when copying over to Windows machine.

**Pull request merged**